### ISSUE TITLE
MSBUILD : error MSB1021: Cannot create an instance of the logger. Could not load file or assembly 'Microsoft.Web.XmlTransform, Version=2.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

### ISSUE THREAD

**Issue by StewartScottRogers** [fsharp/FAKE#1221]
After upgrade to 4.24.3 and then 4.25.2 for FakeLib.dll the currently working build 
Target "BuildApplication" (fun _ ->
   trace "Building Application..."
   !! @".\EncryptionService.sln"
     |> MSBuildRelease artifactsBuildDir "Build"
     |> Log "AppBuild-Output: "
)

Started throwing the following trace… I don’t think “'Microsoft.Web.XmlTransform, Version=2.1.0.0” is even available to download. Please think about using “Pre-release”. I want to trust your updates without being downed. 

"Building ArchivingServices ...."
"H:\tfs\Encryption Service\GlobalBuild.bat"..\
Checking for updates from https://www.nuget.org/api/v2/.
Currently running NuGet.exe 3.3.0.
NuGet.exe is up to date.
Building project with version: LocalBuild
Shortened DependencyGraph for Target Default:
<== Default
   <== BuildApplication
      <== Clean

The resulting target order is:
- Clean
- BuildApplication
- Default
  Starting Target: Clean 
  Cleanup...
  Deleting contents of Artifacts\
  Creating H:\tfs\Encryption Service\Artifacts\Build\
  Finished Target: Clean
  Starting Target: BuildApplication (==> Clean)
  Building Application...
  Building project: H:\tfs\Encryption Service\EncryptionService.sln
  H:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe  "H:\tfs\Encryption Service\EncryptionService.sln" /t:Build /m      /p:RestorePackages="False" /p:OutputPath="H:\tfs\Encryption Service\Artifacts\Build" /p:Configuration="Release" /logger:Fake.MsBuildLogger+ErrorLogger,"H:\tfs\Encryption Service\packages\FAKE.Lib.4.25.2\lib\net451\FakeLib.dll"
  H:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe  "H:\tfs\Encryption Service\EncryptionService.sln" /t:Build /m      /p:RestorePackages="False" /p:OutputPath="H:\tfs\Encryption Service\Artifacts\Build" /p:Configuration="Release" /logger:Fake.MsBuildLogger+ErrorLogger,"H:\tfs\Encryption Service\packages\FAKE.Lib.4.25.2\lib\net451\FakeLib.dll"
  Microsoft (R) Build Engine version 14.0.25123.0
  Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1021: Cannot create an instance of the logger. Could not load file or assembly 'Microsoft.Web.XmlTransform, Version=2.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
Switch: Fake.MsBuildLogger+ErrorLogger,H:\tfs\Encryption Service\packages\FAKE.Lib.4.25.2\lib\net451\FakeLib.dll
Running build failed.
Error:
Building H:\tfs\Encryption Service\EncryptionService.sln failed with exitcode 1.

---
## Build Time Report

Target     Duration

---

Clean      00:00:00.0031121
Total:     00:00:00.3678226

**Comment by forki** [comment]
Can you please find the exact version where this breaks?

**Comment by StewartScottRogers** [comment]
The first breakage happened when I upgraded to  4.24.3 and then again when I upgraded to 4.25.2 this morning. The "FakeLib.dll" says it can't find 'Microsoft.Web.XmlTransform, Version=2.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' . The Solution .sln builds fine without Fake.

I can't find the version of Microsoft.Web.XmlTransform, Version=2.1.0.0 on nuget, but I thought I saw it yesterday when I spent time trying to get the build up. I have two other almost identical projects that I obviously have not yet upgraded the version of FAKE that are still working fine... It started in  Fake "4.24.3" I believe the older version of Microsoft.Web.XmlTransform, Version=2.1.0.0 was something like 1.?.?.?  worked. I think you may have a version that was pulled by the author sometime last night.

**Comment by forki** [comment]
What I meant is: what is the last version that still works for you? This
will make it easier for us to understand the issue.

**Comment by matthid** [comment]
This XmlTransform thing seems to be a constant root of trouble: https://github.com/fsharp/FAKE/search?q=XmlTransform&type=Issues&utf8=%E2%9C%93

**Comment by 0x53A** [comment]
+1
The error started after updating from 4.43 to 4.46

**Comment by sriv** [comment]
TLDR:

Fake.Lib v 4.64.11 + Fake.MsBuildLogger+ErrorLogger requires the below version of dependencies:

```
nuget FSharp.Core 4.2.3
nuget FAKE.Lib
nuget Microsoft.Web.Xdt 2.1.1
```

I faced the same issue. Here's what I found.

MSBUILD: `Microsoft (R) Build Engine version 14.0.25420.1`

Upgraded `Fake.Lib` from `4.63.2` to `4.64.11`.

Started seeing error:

```
MSBUILD : error MSB1021: Cannot create an instance of the logger. Could not load file or assembly 'Microsoft.Web.XmlTransform, Version=2.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
```

I noticed that `paket update` had bumped up `Microsoft.Web.Xdt` to `2.1.3` which contained a different verion of `Microsoft.Web.XmlTransform`.

Pinned `Microsoft.Web.Xdt` to `2.1.1`, this error goes away, here's the next error:

```
MSBUILD : error MSB1021: Cannot create an instance of the logger. Could not load file or assembly 'FSharp.Core, Version=4.4.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
```

Pinned `FSharp.Core` to `4.2.3`.

Problem goes away.

Note: The pinned version of dependencies are what I had earlier (when things were working), it is possible that a newer version of `FSharp.Core` also helps, but I didn't check.

My project is here: https://github.com/getgauge/gauge-csharp

I use Fake + paket.

**Comment by matthid** [comment]
This should in theory be solved with fake 5 as we no longer use msbuild-loggers like this.
